Feat(contracts): Add script + guide to set aggregator address (#1487)
yetanotherco/aligned_layer: docs/0_internal/setting_aggregator_address_multisig.md

# Propose the Transaction for Setting Aggregator Address using Multisig

> If you want to unpause the contracts, you can propose the unpause transaction using the multisig wallet.

Reviewer: ```suggestion
If you want to set the Aggregator Address, you can propose the set Aggregator Address transaction using the multisig wallet.
```

## Prerequisites

- You need to have deployed the contracts following the [Deploy Contracts Guide](./2_deploy_contracts.md).

> ## Propose transaction for UnPause AlignedLayerServiceManager

Reviewer: ```suggestion
## Propose transaction for Set Aggregator Address
```

1. Go to [Safe](https://app.safe.global/home)

2. Click on `New transaction` -> `Transaction Builder`

   ![New transaction](./images/set_aggregator_address_1.png)

   ![Transaction Builder](./images/set_aggregator_address_2.png)

3. . Get the `AlignedLayerServiceManager` address from ```contracts/script/output/mainnet/alignedlayer_deployment_output.json``` or ```contracts/script/output/holesky/alignedlayer_deployment_output.json``` or ```contracts/script/output/sepolia/alignedlayer_deployment_output.json```

4. Paste the `AlignedLayerServiceManager` address on `Enter Address or ENS Name`

   ![Enter Address](./images/set_aggregator_address_3.png)

5. As this is a Proxy contract, choose `Use Implementation ABI`

   ![Use Implementation ABI](./images/set_aggregator_address_4.png)

6. In `contract method selector` choose `setAggregator()` and within `_alignedAggregator(address)` enter the ethereum address of the aggregator.

   ![Choose pause](./images/set_aggregator_address_5.png)

7. Click on `+ Add new transaction`

   You should see the new transaction to be executed

8. Click on `Create batch` to create the transaction.

>    ![Choose pause](./images/set_aggregator_address_6.png)

Reviewer: ```suggestion
   ![Set Aggregator](./images/set_aggregator_address_6.png)
```

>    ![Choose pause](./images/set_aggregator_address_5.png)
> 
> 7. Click on `+ Add new transaction`
> 
>    You should see the new transaction to be executed
> 
> 8. Click on `Create batch` to create the transaction.
> 
>    ![Set Aggregator](./images/set_aggregator_address_6.png)

Reviewer: ```suggestion
   ![Choose set_aggregator](./images/set_aggregator_address_5.png)

7. Click on `+ Add new transaction`

   You should see the new transaction to be executed

8. Click on `Create batch` to create the transaction.

   ![create batch](./images/set_aggregator_address_6.png)
```
